pkg/providers/vsphere/virtualmachine/snapshot.go

```diff
@@ -218,37 +216,37 @@ func GetSnapshotSize(logger logr.Logger,
 	}
 
 	if snapshot == nil {
-		return 0, fmt.Errorf("snapshot with reference %q"+
+		return 0, 0, fmt.Errorf("snapshot with reference %q"+
 			" not found in vmLayout.Snapshot", vmSnapshot.Value)
 	}
 
-	fcdDeviceKeySet := getFCDDeviceKeySet(moVM)
+	fcdDeviceKeySet, err := getFCDDeviceKeySet(moVM)
+	if err != nil {
+		return 0, 0, err
+	}
+
 	fileKeyMap := make(map[int32]int64)
 	for _, file := range vmLayout.File {
 		fileKeyMap[file.Key] = file.Size
 	}
 
-	var total int64
+	var (
+		total    int64
+		fcdTotal int64
+	)
 
 	// Add the file key for the snapshot memory (.vmem) file if present.
 	if snapshot.MemoryKey != -1 {
 		logger.V(5).Info("Adding memoryKey", "memoryKey", snapshot.MemoryKey)
 		total += fileKeyMap[snapshot.MemoryKey]
 	}
 
-	logger.V(5).Info("Adding the file key for snapshot (.vmsn) file", "dataKey", snapshot.DataKey)
+	logger.V(5).Info("Adding the file key for snapshot (.vmsn) file",
```

Comment:
Don't use special chars in log entries. Get rid of the `(.vmsn)` and change to `vmsn`.